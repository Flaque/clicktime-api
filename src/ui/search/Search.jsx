import React from 'react';
import Autocomplete from 'react-autocomplete';
import {matchTerm} from './util.js'

class Search extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      value: ''
    }
    this.onSelect = this.onSelect.bind(this)
  }

  renderItem(item, isHighlighted) {
    return (
      <div
        className = {isHighlighted
            ? 'searchItem searchItem--highlighted'
            : 'searchItem'}
        key = {item.id}
        id = {item.id}
      >{item.name}</div>
    )
  }

  onSelect(value, item) {
    this.setState({ value })
    this.props.onSelect(item.id)
  }

  render() {
    return (
      <div className="search">
        <Autocomplete
          ref = "autocomplete"
          value = {this.state.value}
          items = {this.props.items}
          getItemValue = { (item) => item.name }
          renderItem = {this.renderItem}
          shouldItemRender={matchTerm}
          onChange = { (event, value) => {
            this.setState({ value })
          }}
          onSelect = {this.onSelect}
        />
      </div>
    )
  }
}

export default Search;
