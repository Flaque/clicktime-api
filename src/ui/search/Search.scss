.searchItem {
  cursor: pointer;
  padding: 5px;
  padding-left: 10px;

  &:hover {

  }

  &.searchItem--highlighted {

  }
}
