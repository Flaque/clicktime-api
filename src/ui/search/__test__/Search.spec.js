import React from 'react';
import Search from '../Search.jsx';
import renderer from 'react-test-renderer';

const mockFunction = () => { return }
const mockItems = [{name: 'blah', id: '123'}, {name: 'ya', id: 'jioe'}]

test('Test that Search is still the same', () => {
  const tree = renderer.create(
    <Search items={mockItems} onSelect={ mockFunction } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
