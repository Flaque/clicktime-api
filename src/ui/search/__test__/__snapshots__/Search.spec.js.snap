exports[`test Test that Search is still the same 1`] = `
<div
  className="search">
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }>
    <input
      aria-autocomplete="list"
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      placeholder="Type a task!"
      role="combobox"
      value="" />
  </div>
</div>
`;
